### DISCUSSION TITLE
Multienvioriment test fails when 2nd spec starts.

### DISCUSSION THREAD

**Discussion by Adolfo-Macias** [cypress-io/cypress#16023] [Questions and Help]
**The problem**
I have a code that was designed to run only in one environment and It was working fine when running single spec or multiple spec with a single command, then I was request to run the same code in multiple environment (dev, int, stage).
After the changes describe below:
Running single spec works fine all of them. 
Running multiple spec in a single command the second spec (.js file) fails.

Why running single test works and not when I run all of them?
What I can do to make this works?

**I did some modifications in files to support multiple environments**
Cypress/plugins/index.js
const fs = require('fs-extra')
const path = require('path')
module.exports = (on, config) => {
  function getConfigurationByFile (file) {
    const pathToConfigFile = path.resolve( 'cypress','config', `${file}.json`)
    return fs.readJson(pathToConfigFile)
  }
  on('task', {
    console_log (message) {
      console.log(message)
      return null
    }
  })
  const file = config.env.env   
  return getConfigurationByFile(file)
}

**Created new configuration files** 
Cypress/config/dev.json
Cypress/config/stage.json
Cypress/config/int.json
{
    "baseUrl": "http://page-int.com/",
    "env": {
      "Pub_data": "Publications/pub-int.json",
      "Config": "../../TestData/config-int.json"
    }
}

**Created new test data and configuration for each env** 
Cypress/fixtures/Publications/pub-dev.json
Cypress/fixtures/Publications/pub-stage.json
Cypress/fixtures/Publications/pub-int.json
{
    "Pub_1": {
        "title": "My pub 1 for int",
    },
    "Pub_2":{
        "title": "My pub 2 for int",
    }
}

TestData/config-dev.json
TestData/config-stage.json
TestData/config-int.json
{
  "SsoUser" : "USER",
  "ssoPw" : "PWD"
}

test1.js
describe('test1', function () {
    beforeEach(function () {
        cy.fixture(Cypress.env('Pub_data')).as('allpubs')
        cy.fixture(Cypress.env('Config')).as('Config')
    })
    it('Login', function () {
        cy.visit('/');
        loginSSO(`${this.Config.SsoUser}`, `${this.Config.SsoPw}`);
    })
})
test2.js
describe('test2', function () {
    beforeEach(function () {
        cy.fixture(Cypress.env('Pub_data')).as('allpubs')
        cy.fixture(Cypress.env('Config')).as('Config')
    })
    it('Login', function () {
        cy.visit('/');
        loginSSO(`${this.Config.SsoUser}`, `${this.Config.SsoPw}`);
    })
})

If I run each spec (test1.js and test2.js) separately works fine:
cypress run  --env env=int --spec=cypress\\integration\\Suites\\test1.js
cypress run  --env env=int --spec=cypress\\integration\\Suites\\test2.js

If I run both spec with a single command the first test always pass and the second test always fails.
cypress run  --env env=int --spec=cypress\\integration\\Suites\\*

**Error**
  Running: Suites\test2.js...                                    (2 of 2)

  test2
(node:29212) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 pipe listeners added. Use emitter.setMaxListeners() to increase limit
    1) Login

  0 passing (15s)
  1 failing

  1) test2
       Login:
     CypressError: cy.visit() failed trying to load:

http://page-int.com/

We attempted to make an http request to this URL but the request failed without a response.

We received this error at the network level:

  > Error: Exceeded maxRedirects. Probably stuck in a redirect loop http://page-int.com/oauth/login?code=LONGSTRINGEYEs9LDioyYbrBmBPjr3CY9BjQZGcw24CqjbT80J0z0VvgxkJ2IbnOWwOGClFoDe

Common situations why this would fail:
  - you don't have internet access
  - you forgot to run / boot your web server
  - your web server isn't accessible
  - you have weird network configuration settings on your computer

**Expected behaviour**
All the spec should be executed successfully.

**Desktop (please complete the following information):**

OS: Window 10 Pro
Cypress:    7.0.0                                                                              │
Browser:    Electron 89 (headless)        
     
**Testing notes**
Same behavior with:
Cypress 4.0.0
Cypress 5.0.0
Cypress 7.0.0

I do the SSO using the UI, I know its not best practice
SSO.js

export default function loginSSO(user, pwd) {
  cy.location().then((loc) => {
    if (loc.href.indexOf("login") > -1) {
      cy.task("console_log", 'Logging into SSO')
      cy.get('#sso_username', { timeout: 60000 }).type(user)
      cy.get('#ssopassword', { timeout: 1000 }).type(pwd)
      cy.get('span > input', { timeout: 1000 }).click()
    }
    else{
      cy.task('console_log', 'Skipping SSO')
    }
    
  })
  Cypress.on('uncaught:exception', (err, runnable) => {
    return false
  })
}

**Additional context**

**Comment by bahmutov** [comment]
Could you please take the code and place it in the public repository so we can clone and run it?
